langkit/core/metric.py

```diff
@@ -185,7 +185,7 @@ def _build_metrics(self, modules: List[MetricCreator]) -> List[Metric]:
                 for m in module:
                     if callable(m):
                         schema = m()
-                        if isinstance(schema, Metric):
+                        if isinstance(schema, SingleMetric) or isinstance(schema, MultiMetric):
```

Comment:
Was this a 3.8 type error thing?

Reply:
yes. That is one way to fix it, but maybe there are better solutions?

Reply:
No I think you're right , sadly. I don't think Union could be used this way and I just wasn't using 3.8 locally. We'll have to add a 3.8 CI option to make sure this gets caught.